Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -0,0 +1,544 @@
+    {
+        try
+        {
+            using var timeoutToken = new CancellationTokenSource();
+            timeoutToken.CancelAfter(TimeSpan.FromSeconds(LONG_OPERATION_TIMEOUT_SECONDS));
+            using var combinedTokens =
+                CancellationTokenSource.CreateLinkedTokenSource(timeoutToken.Token, cancellationToken);
+
+            using var requestResources = CreateRequest(string.Empty, HttpMethod.Get);
+
+            requestResources.RequestMessage.RequestUri = new Uri(await GetFileId(remotename));
+
+            await requestResources.HttpClient.DownloadFile(requestResources.RequestMessage, output, null,
+                timeoutToken.Token);
+        }
+        catch (HttpRequestException wex)
+        {
+            /*
+             * Known Behaviour on pCloud
+             *
+             * If the temporary link is no longer valid, it will return a 401
+             * if the url is corrupted/wrong it will return a 500
+             */
+
+            switch (wex)
+            {
+                case { StatusCode: HttpStatusCode.Unauthorized }:
+                    throw new FileMissingException("Temporary link is no longer valid", wex);
+                case { StatusCode: HttpStatusCode.InternalServerError }:
+                    throw new FileMissingException("Temporary link is corrupted", wex);
+                default:
+                    throw;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Delete remote file if it exists, if now, throws FileMissingException
+    /// </summary>
+    /// <param name="remotename">filename to be deleted on the remote</param>
+    /// <param name="cancellationToken">CancellationToken that is combined with internal timeout token</param>
+    /// <returns></returns>
+    /// <exception cref="FileMissingException">FileMissingException when file is not found</exception>
+    /// <exception cref="Exception">Exceptions arising from either code execution or business logic when return code from pcloud indicates an error.</exception>
+    public Task DeleteAsync(string remotename, CancellationToken cancellationToken)
+    {
+        if (!String.IsNullOrWhiteSpace(_Path))
+            remotename = $"/{_Path}/{remotename}";
+
+        using var timeoutToken = new CancellationTokenSource();
+        timeoutToken.CancelAfter(TimeSpan.FromSeconds(SHORT_OPERATION_TIMEOUT_SECONDS));
+        using var combinedTokens =
+            CancellationTokenSource.CreateLinkedTokenSource(timeoutToken.Token, cancellationToken);
+
+        using var requestResources = CreateRequest($"/deletefile?path={remotename}", HttpMethod.Get);
+
+        using var response = requestResources.HttpClient.SendAsync(
+            requestResources.RequestMessage,
+            HttpCompletionOption.ResponseContentRead, combinedTokens.Token).GetAwaiter().GetResult();
+
+        response.EnsureSuccessStatusCode();
+
+        var content = response.Content.ReadAsStringAsync(combinedTokens.Token).GetAwaiter().GetResult();
```

Comment:
Should use `await` instead of `GetAwaiter()`

Reply:
I went with the .Await() extension here.